### PR TITLE
feat: enforce target network

### LINKED ISSUE #250: Connectors popup being cut on bottom
Image
Image

### PR DESCRIPTION
**Pull request by arthurgeron**
# Changes

# @fuels/react
- New FuelChainProvider, meant to be used to enforce the same network across usage of Native connector (Closes https://github.com/FuelLabs/fuel-explorer/issues/523)
> Usage example can be found seen in `examples/react-app`
- Fixed dialogs clipping in mobile devices (Closes #250)

# Other
- Moved icons to a shared folder (i.e. `src/icons`)
- Moved `DialogContent` content to `src/ui/Dialog/components/Content`

image

### CODE REVIEW

File: packages/react/src/providers/FuelProvider.tsx [hunk 1]
```diff
@@ -11,15 +13,22 @@ export { useConnectUI } from './FuelUIProvider';
 type FuelProviderProps = {
   ui?: boolean;
   fuelConfig?: FuelConfig;
+  /**
+   * Whether enforce connectors to be on the desired the network. Requires a FuelChainProvider wrapper.
+   * @default true
+   */
+  monitorNetwork?: boolean;
```
Comment (luizstacio) [inline packages/react/src/providers/FuelProvider.tsx:20]: I would change this config to receive:

- provider | providerUrl | number
- If DApp configures a provider request to be the same one on the wallet check by chainId only.

And include FuelChainProvider inside the FuelUIProvider no need to a new provider in this case.

**Review by luizstacio — CHANGES_REQUESTED** [review]
I believe this change also requires a change on the connectors for all of them to support receive a custom provider.

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
it's missing to check changes from wallet. if I change to a unsupported network directly inside of my wallet, the app is not detecting it

**Review by luizstacio — APPROVED** [review]
LGTM

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**